### DISCUSSION TITLE
Passing filtered queryset disables `DELETE_INACTIVE_DEVICES` option

### DISCUSSION THREAD

**Discussion by karambaq** [xtrinch/fcm-django#251] [General]
If you pass a queryset which is filtered with `active=True`, 

Example:

```
FCMDevice.objects.filter(active=True).send_message(...)
```

Then inactive devices is not being deleted because there is none of them in the queryset after marking them as inactive in this line (That's my guess):

https://github.com/xtrinch/fcm-django/blob/master/fcm_django/models.py#L195

`self.filter(registration_id__in=deactivated_ids).update(active=False)`

I'm not sure if this is expected behavior so making it as discussion. 

Also I have an answer, I'm sending a few messages to the inactive devices, but I don't see them in the `registration_ids_sent` of the firebase response and in `deactivated_registration_ids` too, so they just stands here as inactive. Should I delete them by myself?

**Comment by karambaq** [comment]
@xtrinch

It would be great if you could take the time to clarify the situation 🙏

**Comment by xtrinch** [comment]
@karambaq Yes, they are not getting deleted because you're filtering them when sending. This is expected behaviour.
As for deleting inactive devices, you have a setting called `DELETE_INACTIVE_DEVICES ` which you need to set to true to have them automagically deleted. Otherwise it's up to you to delete them manually

**Reply by karambaq** [reply]
@xtrinch 
Thanks! 

`DELETE_INACTIVE_DEVICES` option deletes a device only when we receive error from Firebase, am I right?
So if there is an inactive device in the database and we are not trying to send notification to it, then it will never be deleted unless we do it manually. 

From firebase documentation I understand that we should delete devices which are more than 2 months old. But I wonder is it right way to delete inactive devices everyday.

**Reply by xtrinch** [reply]
Correct.

For regular pruning you can set up a cronjob. Although I'm not entirely sure of the usefulness of that. Is there a reason you cannot delete upon trying to send?

**Reply by karambaq** [reply]
We faced an issue. Devices receiving a duplicate notifications, was using 0.3.4 version of this library, currently updated to 2.0.0. 
I don't think that inactive devices have any good reason to stay in the database.

**Reply by xtrinch** [reply]
Are you sure you're not registering them twice? Or perhaps not de-registering upon logout or something like that?

**Reply by karambaq** [reply]
Yeah, checked that.
